awpy/visualization/plot.py

```diff
@@ -326,3 +341,382 @@ def plot_nades(
     axes.get_xaxis().set_visible(b=False)
     axes.get_yaxis().set_visible(b=False)
     return figure, axes
+
+
+def _plot_map_control_snapshot_helper(
+    map_name: str,
+    ct_tiles: dict,
+    t_tiles: dict,
+    axes: plt.Axes,
+    player_data: dict | None = None,
+) -> None:
+    """Helper function to plot map control nav tile plot.
+
+    Args:
+        map_name (str): Map used position_transform call
+        ct_tiles (dict): CT map control values dictionary
+        t_tiles (dict): T map control values dictionary
+        axes (plt.axes): axes object for plotting
+        player_data (dict): Dictionary of player positions
+            for each team. Expected format same as output of extract_player_positions
+
+    Returns: Nothing, all plotting is done on ax object
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+                    If player_dict not of expected
+                    extract_player_positions output format
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    ct_tile_set, t_tile_set = set(ct_tiles.keys()), set(t_tiles.keys())
+    all_tiles = ct_tile_set.union(t_tile_set)
+
+    # Iterate through the tiles that have a value
+    for tile in all_tiles:
+        if tile in NAV[map_name]:
+            area = NAV[map_name][tile]
+            se_x = position_transform(map_name, area["southEastX"], "x")
+            nw_x = position_transform(map_name, area["northWestX"], "x")
+            se_y = position_transform(map_name, area["southEastY"], "y")
+            nw_y = position_transform(map_name, area["northWestY"], "y")
+            width = se_x - nw_x
+            height = nw_y - se_y
+
+            # Use max value (if exists) for each side for the current tile
+            ct_val = max(ct_tiles[tile] + [0])
```

Comment:
It's in case there are 0 CT values for that particular tile (which would mean only T has control of that tile). This is a common scenario since a tile could be in T control but not in CT control. So to catch that I aded the [0] in case the dictionary lookup returns an empty list.

Thanks for the * suggestion! Just wondering what's the improvement with the operator compared to just using max on the list?

Reply:
Actually my suggestion doesnt work for the empty list anyway. So your current way is the objectively correct one.

Reply:
lol I was just about to comment what you said cause my notebook failed due to this error but didn't see your latest comment. Decided to use the default argument for the max function to catch empty list if it occurs (see [here](https://stackoverflow.com/questions/36157995/a-safe-max-function-for-empty-lists)). This will be changed in next commit to branch.